uberenv.py

```diff
@@ -826,39 +850,6 @@ def patch(self):
 
         # force spack to use only "defaults" config scope
         self.disable_spack_config_scopes(spack_dir)
-        spack_etc_defaults_dir = pjoin(spack_dir,"etc","spack","defaults")
-
-        if cfg_dir is not None:
-            # copy in "defaults" config.yaml
-            config_yaml = pabs(pjoin(cfg_dir,"..","config.yaml"))
-            if os.path.isfile(config_yaml):
-                sexe("cp {0} {1}/".format(config_yaml, spack_etc_defaults_dir), echo=True)
-            mirrors_yaml = pabs(pjoin(cfg_dir,"..","mirrors.yaml"))
-            if os.path.isfile(mirrors_yaml):
-                sexe("cp {0} {1}/".format(mirrors_yaml, spack_etc_defaults_dir), echo=True)
-
-            # copy in other settings per platform
-            print("[copying uberenv compiler and packages settings from {0}]".format(cfg_dir))
-
-            config_yaml    = pjoin(cfg_dir,"config.yaml")
-            mirrors_yaml   = pjoin(cfg_dir,"mirrors.yaml")
-            compilers_yaml = pjoin(cfg_dir,"compilers.yaml")
-            packages_yaml  = pjoin(cfg_dir,"packages.yaml")
-
-            if os.path.isfile(config_yaml):
-                sexe("cp {0} {1}/".format(config_yaml , spack_etc_defaults_dir ), echo=True)
-
-            if os.path.isfile(mirrors_yaml):
-                sexe("cp {0} {1}/".format(mirrors_yaml , spack_etc_defaults_dir ), echo=True)
-
-            if os.path.isfile(compilers_yaml):
-                sexe("cp {0} {1}/".format(compilers_yaml, spack_etc_defaults_dir ), echo=True)
-
-            if os.path.isfile(packages_yaml):
-                sexe("cp {0} {1}/".format(packages_yaml, spack_etc_defaults_dir ), echo=True)
-        else:
-            # let spack try to auto find compilers
-            sexe("{0} compiler find".format(self.spack_exe_path()), echo=True)
 
         # hot-copy our packages into spack
         if len(self.packages_paths) > 0:
```

Comment:
This section should be augmented to only copy if the packages directory is not a spack repo. If it is a spack repo it should call `spack repo add /path`.

Reply:
I was keeping that one for a later issue... 😔

Reply:
im having to move this from patch() and into a new function after the spack environment has been created

Reply:
to determine if there is a spack package repo, i scan each package path in `spack_packages_path` and check if there is a `repo.yaml` in the previous directory.

if i should just check `package_source_dir` instead, let me know.

Reply:
Actually @white238 I’m really thinking that this should be addressed in a subsequent PR: we don’t have that "feature" yet in Uberenv, and we will only delay the merge of this one. Additionally, my personal take on this is that we could/should _require_ that custom packages be defined in a legit spack repo and definitively stop hot copying packages.

Reply:
Scrolling down and I see there is already a commit for it :face_palming:.

Reply:
I agree we should require it. There have been more times than I care to admit that I editted the package file in the repo and reran the specific spack command only to remember it wasn't gonna get copied.

Reply:
I would vote for doing it in this PR because we already have several breaking changes. Might as well just do it in this one.

Reply:
As long as this works seamlessly. How confident are we that wont open up a new surface of confusion?

Hot copying is dead simple to understand and one of the reasons that uberenv has been flexible despite a host of changes in spack proper.

Reply:
I personally think it is only an improvement. I recommend we remove the hot-copying and add an error check for correctness and tells you how to fix it. For example, if there isnt a `repo.yaml` in the correct spot it errors immediately and tells you to create one there.

Reply:
I trust you, but that's also not the question I asked :-).

It would be good to know what the new failure modes are due to this change. 

Documenting those is important b/c it is a big change. 

For example:

We never needed to write a `repo.yaml` before. (More yaml files for the yaml overloads :-))

I am fine with moving forward, but we need be prepared to help folks understand how to move and fix things when things go off the rails b/c we know they will.

Reply:
I agree. I will add a section in the documentation for this